packages/harpguru-core/src/utils/reduce-for-new-column-bounds/reduce-for-new-column-bounds.test.ts

```diff
@@ -0,0 +1,201 @@
+import type { Dispatch } from 'reactn/default'
+import { getHarpStrata } from 'harpstrata'
+import {
+  TuningIds,
+  PitchIds,
+  PozitionIds,
+  ValvingIds,
+  DegreeIds,
+  HarpFaceMatrix,
+  Degree,
+} from 'harpparts'
+
+import { ZoomIds } from '../../types'
+import type { GlobalState } from '../../types'
+
+import { reduceForNewColumnBounds } from './reduce-for-new-column-bounds'
+
+// TODO: Writing this file has made me realise 2 things:
+//
+// 1. columnBounds should perhaps be called holeBounds
+// 2. there is more functionality here than we need in this dispatcher. The UI should make sure that there are no 7 hole zoom requests to a harp which has less than 7 holes. However this functionality will be important for the callback which runs when the activeHarpStrata is updated to update the columnBounds. I might just house all that functionality under this one function for now in anticipation of trying to have a single function to handle all the situations. I don't know
```

Comment:
Point 2 is addressed now. I'm happy with how this is working for both directions of update:

1. Updating the columnBounds due to new zoomId
2. Updating the columnBounds becuase a new harpstrata has been selected

Reply:
Both comments have been rendered redundant actually. See https://github.com/js-jslog/harpguru/pull/115/commits/7722b26f2cb96d63d8955c5f41c8f7c41f5a72d3 for details.